server/src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -913,13 +1003,7 @@ public synchronized XContentBuilder toXContent(XContentBuilder builder, Params p
             builder.field(Fields.REUSED, reusedFileCount());
             builder.field(Fields.RECOVERED, recoveredFileCount());
             builder.field(Fields.PERCENT, String.format(Locale.ROOT, "%1.1f%%", recoveredFilesPercent()));
-            if (params.paramAsBoolean("details", false)) {
```

Comment:
i wonder if it makes sense to do a bwc test for the serialisation.

Reply:
If it's easy to do, It would be great I think, although I see that the version is used when reading/writing:
https://github.com/crate/crate/pull/13187/files#diff-ca3e9db31cc67e6eb2c473cc26e9f7e3843dfc964234f64add8ef02891cf5a56R706
https://github.com/crate/crate/pull/13187/files#diff-ca3e9db31cc67e6eb2c473cc26e9f7e3843dfc964234f64add8ef02891cf5a56R724
@seut What do you think?

Reply:
Looks like `RecoveryState.toXContent(...)` is not used at all at CrateDB? If so lets clean this class upfront..

Reply:
Thanks for the feedback @seut